crates/circuit/src/packed_instruction.rs

```diff
@@ -848,3 +957,23 @@ impl PackedInstruction {
         }
     }
 }
+
+// When cloning a `PackedInstruction` we will need to delete the old cached
+// operation to avoid having duplicated references to the same python object.
+impl Clone for PackedInstruction {
+    fn clone(&self) -> Self {
+        Self {
+            op: self.op.clone(),
+            qubits: self.qubits,
+            clbits: self.clbits,
+            params: self.params.clone(),
+            label: self.label.clone(),
+            #[cfg(feature = "cache_pygates")]
+            py_op: OnceLock::new(),
```

Comment:
@raynelfss I think it's probably worth doing this. Maybe something like this (untested):

```suggestion
            py_op: if matches!(self.op.view(), OperationRef::StandardGate(_))
                && self.params.is_none() && self.label.is_none() {
                self.py_op.clone()
            } else {
                OnceLock::new(),
            }
```